src/SourceBuild/content/eng/tools/PrBaselinePublisher/Publisher.cs

```diff
@@ -0,0 +1,378 @@
+        }
+        else if (pipeline == Pipelines.License)
+        {
+            testResultsTreeItems = await UpdateLicenseScanFilesAsync(updatedTestsFiles, testResultsTreeItems);
+        }
+        else
+        {
+            string errorMessage = "Invalid pipeline.";
+            Log.LogError(errorMessage);
+            throw new InvalidOperationException(errorMessage);
+        }
+
+        var testResultsTreeResponse = await CreateTreeFromItemsAsync(testResultsTreeItems);
+        var parentTreeResponse = await CreateParentTreeAsync(testResultsTreeResponse, originalTreeResponse, originalTestResultsPath);
+
+        await CreateOrUpdatePullRequestAsync(parentTreeResponse, buildId, title, targetBranch);
+
+        return Log.GetExitCode();
+    }
+
+    // Return a dictionary using the filename without the 
+    // "Updated" prefix and anything after the first '.' as the key
+    private Dictionary<string, HashSet<string>> GetUpdatedFiles(string updatedTestsResultsPath) =>
+        Directory
+            .GetFiles(updatedTestsResultsPath, "Updated*", SearchOption.AllDirectories)
+            .GroupBy(updatedTestsFile => ParseUpdatedFileName(updatedTestsFile).Split('.')[0])
+            .ToDictionary(
+                group => group.Key,
+                group => new HashSet<string>(group)
+            );
+
+    private async Task<List<NewTreeItem>> UpdateLicenseScanFilesAsync(Dictionary<string, HashSet<string>> updatedFiles, List<NewTreeItem> tree)
+    {
+        foreach (var updatedFile in updatedFiles)
+        {
+            if (updatedFile.Key.Contains("Exclusions"))
+            {
+                // Combine the exclusions files
+                IEnumerable<string> parsedFile = Enumerable.Empty<string>();
+                foreach (var filePath in updatedFile.Value)
+                {
+                    var updatedFileLines = File.ReadAllLines(filePath);
+                    parsedFile = parsedFile.Any() ? parsedFile.Where(parsedLine => updatedFileLines.Contains(parsedLine)) : updatedFileLines;
+                }
+                string? content = parsedFile.Any() ? string.Join("\n", parsedFile) + "\n" : null;
+                string updatedFilePath = updatedFile.Key + ".txt";
+                tree = await UpdateFileAsync(tree, content, updatedFile.Key, updatedFilePath);
+            }
+            else
+            {
+                // Update the other files
+                foreach (var filePath in updatedFile.Value)
+                {
+                    var content = File.ReadAllText(filePath);
+                    if (content == DefaultLicenseBaselineContent)
+                    {
+                        content = null;
+                    }
+                    string originalFileName = Path.GetFileName(ParseUpdatedFileName(filePath));
+                    string updatedFilePath = Path.Combine("baselines/licenses", originalFileName);
```

Comment:
I wasn't expecting the baselines/licenses would have to be hardcoded here.  I thought this could be parted from the OriginalTestResultsPath

Reply:
This is unfortunately something that I couldn't quite find another way around. The issue comes from the fact that the LicenseExclusions.txt file exists in OriginalTestResultsPath, but the baselines exist in OriginalTestResultsPath/baselines/licenses, and there isn't really a way to "look ahead" for the file in OriginalTestResultsPath/baselines/licenses if it doesn't exist (which sometimes the baselines do not previously exist).

A longterm solution to this problem would be to have all like test files exist in the same directories. For example, license exclusions and baselines would live in one directory, and all sdk content exclusions and baselines would live in another.

Reply:
There's nothing special about the current layout of exclusion and baseline files. If changing that layout simplifies this tool, let's do that.

Reply:
Fixing in https://github.com/dotnet/installer/pull/19650